BLL/Services/DirectoryService.cs

```diff
@@ -88,19 +101,41 @@ public CityDTO GetCityById(int cityId)
             return _mapper.Map<City, CityDTO>(_db.Cities.Get(cityId));
         }
 
+        public string GetLocationById(int cityId)
+        {
+            var city = _db.Cities.Get(cityId);
+            var state = _db.States.Get(city.State_Id);
+            var country = _db.Countries.Get(state.Country_Id);
+            return $"{country.CountryName}, {state.StateName}, {city.CityName}";
+        }
+
         public List<CityDTO> GetAllCities()
         {
             return _mapper.Map<List<City>, List<CityDTO>>(_db.Cities.GetAll().ToList());
         }
 
-        //public StatusDTO GetStatusById(int statusId)
-        //{
-        //    return _mapper.Map<Status, StatusDTO>(_db.Statuses.Get(statusId));
-        //}
+        public List<CityDTO> GetAllCitiesByCountryId(int countryId)
+        {
+            var statesInCountry = _db.States.GetAll().Where(x => x.Country_Id == countryId).ToList();
```

Comment:
it's very bad approach
the whole method can be replaced with single code line:

return _mapper.Map<List<City>, List<CityDTO>>(_db.Cities.FindWithSpecificationPattern(new CitySpecification()).Where(x => x.State.Country_Id == countryId).ToList());

all you need is just to include State and State.Country in the CitySpecification class